internal/provider/managed_radius/model.go

```diff
@@ -0,0 +1,113 @@
+package managed_radius
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	v20250101 "github.com/smallstep/terraform-provider-smallstep/internal/apiclient/v20250101"
+	"github.com/smallstep/terraform-provider-smallstep/internal/provider/utils"
+)
+
+const name = "smallstep_managed_radius"
+
+type ManagedRadiusModel struct {
+	ID              types.String `tfsdk:"id"`
+	Name            types.String `tfsdk:"name"`
+	NASIPs          types.List   `tfsdk:"nas_ips"`
+	ClientCA        types.String `tfsdk:"client_ca"`
+	ReplyAttributes types.List   `tfsdk:"reply_attributes"`
+
+	ServerCA       types.String `tfsdk:"server_ca"`
+	ServerIP       types.String `tfsdk:"server_ip"`
+	ServerPort     types.String `tfsdk:"server_port"`
+	ServerHostname types.String `tfsdk:"server_hostname"`
+}
+
+type ReplyAttributeModel struct {
+	Name               types.String `tfsdk:"name"`
+	Value              types.String `tfsdk:"value"`
+	ValueFromExtension types.String `tfsdk:"value_from_extension"`
+}
+
+var replyAttributeTypes = map[string]attr.Type{
+	"name":                 types.StringType,
+	"value":                types.StringType,
+	"value_from_extension": types.StringType,
+}
+
+func toAPI(ctx context.Context, diags *diag.Diagnostics, model *ManagedRadiusModel) v20250101.ManagedRadius {
```

Comment:
Although the providers uses the `toAPI` and `fromAPI` methods, when I added the strategies I had to generalize some models, and add a `FromAPI` and a `Model.ToAPI` methods. Something like this

```go
type Model struct{
  ID types.String ``tfsdk:"id"`
  // ...
}

func FromAPI(ctx context.Context, v *v20250101.XXX, state utils.AttributeGetter, root path.Path) (types.Object, diag.Diagnostics) {
  // ...
}

func (m *Model) ToAPI(ctx context.Context, obj types.Object) (*v20250101.XXX, diag.Diagnostics) {
  // ...
}
```

It might not be necessary right now, but if we need to compose multiple models it really helps.